core/store/migrations/migration1587580235/migrate.go

```diff
@@ -0,0 +1,27 @@
+package migration1587580235
+
+import (
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+)
+
+type logConsumption struct {
+	ID           *models.ID  `gorm:"primary_key"`
+	BlockHash    common.Hash `gorm:"not null;unique_index:idx_unique_log_consumption"`
+	ConsumerType string      `gorm:"not null;unique_index:idx_unique_log_consumption"`
+	ConsumerID   *models.ID  `gorm:"not null;unique_index:idx_unique_log_consumption"`
+	LogIndex     uint        `gorm:"not null;unique_index:idx_unique_log_consumption"`
+	CreatedAt    time.Time
```

Comment:
1. Lets pls not use gorm’s automigrate feature any more, it has way too much magic and makes it difficult to see whats actually been created
2. Why choose a string ID?
4. created at should be not null
5. if it has a created_at, should also have updated_at really
6. What is the maximum expected size of logindex?
7. It is not necessary to specify CreatedAt in go world, since gorm writes this automatically
8. Type should probably be enum
9. ConsumerID is a UUID and should be UUID type not bytea
10. ConsumerID looks like a foreign key to job_specs.ID - it should have a constraint